geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -5237,6 +5237,9 @@ public boolean basicBridgePut(Object key, Object value, byte[] deltaBytes, boole
       if (success) {
         clientEvent.setVersionTag(event.getVersionTag());
         getCachePerfStats().endPut(startPut, event.isOriginRemote());
+      } else if (clientEvent.isConcurrencyConflict() && clientEvent.isPossibleDuplicate()
+          && clientEvent.getVersionTag() == null) {
+        clientEvent.setVersionTag(event.getVersionTag());
```

Comment:
Is `stopper.checkCancelInProgress(null);` needed here as well?

Reply:
Good question. I don't think so. It'll fall through to that check if this condition fails.